Sources/ChatImpl.swift

```diff
@@ -813,5 +813,32 @@ extension ChatImpl: Chat {
     }
   }
 
+  public func addConnectionStatusListener(_ listener: @escaping (ConnectionStatus) -> Void) -> AutoCloseable {
+    AutoCloseableImpl(
+      chat.addConnectionStatusListener { kmpConnectionStatus in
+        switch kmpConnectionStatus.category {
+        case .pnConnectionOnline:
+          listener(.online)
+        case .pnConnectionOffline:
+          listener(.offline)
+        case .pnConnectionError:
+          if let exception = kmpConnectionStatus.exception {
+            listener(.connectionError(error: exception.asError()))
+          }
+        default:
+          break
+        }
+      }, owner: self
+    )
+  }
+
+  public func reconnectSubscriptions() {
```

Comment:
Did you consider adding test for reconnectSubscriptions() and disconnectSubscriptions() ?

Reply:
Yes, I already added them